File: NextApp/iOS/OrgPortalNext/Core/Model/Manual.swift
import Foundation

public struct Manual: Equatable, Codable, Sendable {
    public let id: String
    public let communityId: String?
    public let title: String
    public let body: String
    public let sortOrder: Int
    public let imageUrls: [String]
    public let pdfUrl: String?
    public let externalUrl: String?
    public let isPublished: Bool

    public init(
        id: String,
        communityId: String?,
        title: String,
        body: String,
        sortOrder: Int,
        imageUrls: [String] = [],
        pdfUrl: String? = nil,
        externalUrl: String? = nil,
        isPublished: Bool
    ) {
        self.id = id
        self.communityId = communityId
        self.title = title
        self.body = body
        self.sortOrder = sortOrder
        self.imageUrls = imageUrls
        self.pdfUrl = pdfUrl
        self.externalUrl = externalUrl
        self.isPublished = isPublished
    }

    public var listIdentity: String {
        "\(communityId ?? "shared"):\(id)"
    }
}

File: NextApp/Android/core/model/src/main/kotlin/jp/komehyappyo/member/next/core/model/Manual.kt
package jp.komehyappyo.member.next.core.model

data class Manual(
    val id: String,
    val communityId: String?,
    val title: String,
    val body: String,
    val sortOrder: Int,
    val imageUrls: List<String> = emptyList(),
    val pdfUrl: String? = null,
    val externalUrl: String? = null,
    val isPublished: Boolean,
) {
    val listIdentity: String
        get() = "${communityId ?: "shared"}:$id"
}
